operator/src/main/java/org/bf2/operator/managers/IngressControllerManager.java

```diff
@@ -432,6 +449,66 @@ void reconcileIngressControllers() {
         }
     }
 
+    public void ensureBlueprintRouteMatching(Route route, String blueprintBaseName) {
+        var annotations = Optional.ofNullable(route.getMetadata()).map(ObjectMeta::getAnnotations);
+        var tslConfig = Optional.ofNullable(route.getSpec()).map(RouteSpec::getTls);
+
+        ensureBlueprintRouteMatching(annotations, tslConfig, blueprintBaseName);
+    }
+
+    public void ensureBlueprintRouteMatching(Optional<Map<String, String>> annotations, Optional<TLSConfig> tlsConfig, String blueprintBaseName) {
+        // see findMatchingBlueprint https://github.com/openshift/router/blob/master/pkg/router/template/configmanager/haproxy/manager.go#L817
+        if (!Boolean.TRUE.equals(dynamicConfigManager) || blueprintRouteNamespace == null) {
+            return;
+        }
+
+        var orderedAnnotations = annotations.map(TreeMap::new);
+        MessageDigest stableIdDigest;
+        try {
+            stableIdDigest = MessageDigest.getInstance("SHA-1");
+            orderedAnnotations.ifPresent(m -> m.forEach((k, v) -> {
+                stableIdDigest.update(String.valueOf(k).getBytes());
+                stableIdDigest.update(String.valueOf(v).getBytes());
+            }));
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
+        }
+        tlsConfig.map(TLSConfig::getTermination).map(String::getBytes).ifPresent(stableIdDigest::update);
+        tlsConfig.map(TLSConfig::getCertificate).map(String::getBytes).ifPresent(stableIdDigest::update);
+        tlsConfig.map(TLSConfig::getKey).map(String::getBytes).ifPresent(stableIdDigest::update);
+        tlsConfig.map(TLSConfig::getCaCertificate).map(String::getBytes).ifPresent(stableIdDigest::update);
```

Comment:
Should we worry about the certificate expiration and renew causing the blueprint hash changed and haproxy reloaded?

Reply:
That's only applicable to `edge` termination and we only use that with the kas-installer.   In production all routes are `passthrough` so haproxy has no knowledge of the certificate. 

I don't think this will cause an issue in development, so I think this is tolerable.

Reply:
Make sense. Thanks.

Reply:
@showuon I'll resolve this comment